File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/Code Sweep/C#,C++/TDD/Mocks/MockServiceProvider.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VisualStudio.Shell.Interop;
using EnvDTE;
using Microsoft.VisualStudio.TextManager.Interop;
using Microsoft.VisualStudio.Shell;
using System.ComponentModel.Design;
using System.Diagnostics;

namespace Microsoft.Samples.VisualStudio.CodeSweep.UnitTests
{
    class MockServiceProvider : IServiceProvider
    {
        readonly MockTaskList _taskList = new MockTaskList();
        readonly MockShell _uiShell = new MockShell();
        readonly MockStatusBar _statusBar = new MockStatusBar();
        readonly MockDTE _dte;
        readonly MockSolution _solution = new MockSolution();
        readonly MockRDT _rdt = new MockRDT();
        readonly MockUIShellOpenDocument _uiShellOpenDoc = new MockUIShellOpenDocument();
        readonly MockTextManager _textMgr = new MockTextManager();
        readonly MockWebBrowsingService _webBrowser = new MockWebBrowsingService();
        readonly OleMenuCommandService _menuService;

        public MockServiceProvider()
        {
            _menuService = new OleMenuCommandService(this);
            _dte = new MockDTE(this);
        }

        public MockTaskList TaskList
        {
            get { return _taskList; }
        }

        #region IServiceProvider Members

        public object GetService(Type serviceType)
        {
            if (typeof(SVsTaskList).IsEquivalentTo(serviceType))
            {
                return _taskList;
            }
            else if (typeof(SVsUIShell).IsEquivalentTo(serviceType))
            {
                return _uiShell;
            }
            else if (typeof(SVsStatusbar).IsEquivalentTo(serviceType))
            {
                return _statusBar;
            }
            else if (typeof(DTE).IsEquivalentTo(serviceType))
            {
                return _dte;
            }
            else if (typeof(SVsSolution).IsEquivalentTo(serviceType))
            {
                return _solution;
            }
            else if (typeof(SVsRunningDocumentTable).IsEquivalentTo(serviceType))
            {
                return _rdt;
            }
            else if (typeof(SVsUIShellOpenDocument).IsEquivalentTo(serviceType))
            {
                return _uiShellOpenDoc;
            }
            else if (typeof(SVsTextManager).IsEquivalentTo(serviceType))
            {
                return _textMgr;
            }
            else if (typeof(SVsWebBrowsingService).IsEquivalentTo(serviceType))
            {
                return _webBrowser;
            }
            else if (typeof(IMenuCommandService).IsEquivalentTo(serviceType))
            {
                return _menuService;
            }
            else if (typeof(ISelectionService).IsEquivalentTo(serviceType))
            {
                return null;
            }
            else if (typeof(IDesignerHost).IsEquivalentTo(serviceType))
            {
                return null;
            }
            else
            {
                Debug.Fail("Service " + serviceType + " not found.");
                return null;
            }
        }

        #endregion
    }
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[C++]-Visual Studio 2010 SDK Samples/Code Sweep/C++,VB/TDD/Mocks/MockServiceProvider.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.VisualStudio.Shell.Interop
Imports EnvDTE
Imports Microsoft.VisualStudio.TextManager.Interop
Imports Microsoft.VisualStudio.Shell
Imports System.ComponentModel.Design
Imports System.Diagnostics

Namespace Microsoft.Samples.VisualStudio.CodeSweep.UnitTests
    Friend Class MockServiceProvider
        Implements IServiceProvider
        Private ReadOnly _taskList As New MockTaskList()
        Private ReadOnly _uiShell As New MockShell()
        Private ReadOnly _statusBar As New MockStatusBar()
        Private ReadOnly _dte As MockDTE
        Private ReadOnly _solution As New MockSolution()
        Private ReadOnly _rdt As New MockRDT()
        Private ReadOnly _uiShellOpenDoc As New MockUIShellOpenDocument()
        Private ReadOnly _textMgr As New MockTextManager()
        Private ReadOnly _webBrowser As New MockWebBrowsingService()
        Private ReadOnly _menuService As OleMenuCommandService

        Public Sub New()
            _menuService = New OleMenuCommandService(Me)
            _dte = New MockDTE(Me)
        End Sub

        Public ReadOnly Property TaskList() As MockTaskList
            Get
                Return _taskList
            End Get
        End Property

#Region "IServiceProvider Members"

        Public Function GetService(ByVal serviceType As Type) As Object Implements IServiceProvider.GetService
            If GetType(SVsTaskList).IsEquivalentTo(serviceType) Then
                Return _taskList
            ElseIf GetType(SVsUIShell).IsEquivalentTo(serviceType) Then
                Return _uiShell
            ElseIf GetType(SVsStatusbar).IsEquivalentTo(serviceType) Then
                Return _statusBar
            ElseIf GetType(DTE).IsEquivalentTo(serviceType) Then
                Return _dte
            ElseIf GetType(SVsSolution).IsEquivalentTo(serviceType) Then
                Return _solution
            ElseIf GetType(SVsRunningDocumentTable).IsEquivalentTo(serviceType) Then
                Return _rdt
            ElseIf GetType(SVsUIShellOpenDocument).IsEquivalentTo(serviceType) Then
                Return _uiShellOpenDoc
            ElseIf GetType(SVsTextManager).IsEquivalentTo(serviceType) Then
                Return _textMgr
            ElseIf GetType(SVsWebBrowsingService).IsEquivalentTo(serviceType) Then
                Return _webBrowser
            ElseIf GetType(IMenuCommandService).IsEquivalentTo(serviceType) Then
                Return _menuService
            ElseIf GetType(ISelectionService).IsEquivalentTo(serviceType) Then
                Return Nothing
            ElseIf GetType(IDesignerHost).IsEquivalentTo(serviceType) Then
                Return Nothing
            Else
                Debug.Fail("Service " & CType(CType(serviceType, Object), String) & " not found.")
                Return Nothing
            End If
        End Function

#End Region
    End Class
End Namespace
